modules/core/03-connection/keeper/verify.go

```diff
@@ -98,6 +25,10 @@ func (k *Keeper) VerifyConnectionState(
 	connectionID string,
 	counterpartyConnection types.ConnectionEnd, // opposite connection
 ) error {
+	if !types.IsValidConnectionID(connectionID) {
+		return types.ErrInvalidConnectionIdentifier
+	}
```

Comment:
I'm curious behind the context of adding this check

Reply:
I was debugging some test. I removed it now, thanks for catching. Maybe we should actually add this check regardless?

Reply:
We use it in `ValidateBasic` afaik